File: cpp/24d_1399A_RemoveSmallest.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp, cnt = 0;
    std::vector<std::string> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::unordered_set<int> hset;
        cnt = 0;
        std::cin>>n;
        for(int i = 0; i < n; i++){
            std::cin>>temp;
            hset.insert(temp);
        }
        for(auto i : hset){
            if(hset.find(i-1) == hset.end()){
                cnt++;
            }
        }
        if(cnt == 1){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/24d_1399A_RemoveSmallest.py
def main():
    t = int(input())
    ans = []
    for _ in range(t):
        n = int(input())
        hset = set()
        numbs = input().split()
        for i in numbs:
            hset.add(int(i))
        cnt = 0
        for i in hset:
            if i-1 not in hset:
                cnt += 1
        if cnt == 1:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()